### PR TITLE
Show plugin release date in plugin manager

### PR DESCRIPTION
**Pull request by daniel-beck**
This was one of the feedbacks at the contributor summit at FOSDEM: It would be helpful to see at a glance when a plugin was released, to help understand how actively it is maintained.

Up to date screenshots in https://github.com/jenkinsci/jenkins/pull/4535#issuecomment-593385560

### Proposed changelog entries

* Show plugin release date in plugin manager

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [n/a] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [n/a] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
One comment about UX, label display names should probably come from the update center: https://github.com/jenkins-infra/plugin-site-api/issues/82 .Though it keeps a question about custom update centers then

**Review by StefanSpieker — APPROVED** [review]
That is a nice feature. Thanks 👍

**Comment by timja** [comment]
Alignment of the version column seems wrong on the available tab:

image

It's fine on the installed column, and I don't have any plugin updates atm, but I assume it's fine there as that's what is screenshotted

I was also happy to see that the date is localised and I get the date format that I am used to

**Review by timja — DISMISSED** [review]
See https://github.com/jenkinsci/jenkins/pull/4535#issuecomment-593075067,

should be a minor fix hopefully

**Comment by timja** [comment]
Hmm taking another look without this change,
it already seems to be misaligned but this makes it worse,

Removing my change requested but I still think it should be improved:

image

**Comment by daniel-beck** [comment]
> Alignment of the version column seems wrong on the available tab:

It's due to ridiculously long version numbers like `1.12+build.201809061734` or `3.8.20200204-101107.d1d344b` and left-alignment of the column content. Unrelated to this change.

File: core/src/main/java/hudson/model/UpdateSite.java [hunk 1]
```diff
@@ -1018,6 +1020,12 @@ private static String get(JSONObject o, String prop) {
          */
         private Set<Plugin> incompatibleParentPlugins;
 
+        /**
+         * Date when this plugin was released.
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/UpdateSite.java]:
```suggestion
         * Date when this plugin was released.
         * @since TODO
```

**Review by oleg-nenashev — APPROVED** [review]
I wonder whether formatting would differ depending on the user locale, but it is fine even if it is not

**Comment by daniel-beck** [comment]
> I wonder whether formatting would differ depending on the user locale

It does, that's the whole point of using this control.

**Comment by timja** [comment]
> I wonder whether formatting would differ depending on the user locale, but it is fine even if it is not

if you see my screenshot it has a different locale to DB's

**Comment by daniel-beck** [comment]
This now depends on https://github.com/jenkinsci/jenkins/pull/4534 not being merged before this, or https://github.com/jenkinsci/jenkins/pull/4535/files#diff-6c89c5cced7eb382005a0b0825a45140R70 needs `onclick` removed otherwise. Should be trivial to do on GitHub and ideally not hold off merging for technicalities.

**Review by timja — APPROVED** [review]

**Comment by fqueiruga** [comment]
I know I'm late to the party but I have a suggestion. Can we consider, instead of having another numeric column showing the date, showing the last release as  text? That will remove mental load for the user.

We could also show that message within the version column. The information relates to the version anyway. An example:
Captura de pantalla 2020-03-02 a las 10 24 53

**Comment by daniel-beck** [comment]
> We could also show that message within the version column. The information relates to the version anyway.

I considered that, but was unsure how this would impact sorting by this column. Given how useless sort by version is, I guess this column would then sort by release date? Is that useful UX?

**Comment by fqueiruga** [comment]
If we want to sort by release date, we would definitely need a column just for it IMO

**Comment by daniel-beck** [comment]
> If we want to sort by release date, we would definitely need a column just for it IMO

Well, the table is sortable. Right now it's possible to sort by name and version number (🤣). Sorting by date is potentially the most useful of any of them (even if there's no obvious use case other than something like getting general ecosystem status at a glance).

Anyway, I like your "time span" idea a lot better than just displaying the date. I will implement that. The specific date can always be the tool tip.

**Comment by fqueiruga** [comment]
May I also suggest using the `<time>` tag? With tooltip would look something like this `<time datetime="2019-08-15" tooltip="15/08/2019">8 months ago</time>`.

**Comment by daniel-beck** [comment]
Updated based on feedback. It now looks like this:

Screenshot 2020-03-02 at 13 34 16
Screenshot 2020-03-02 at 13 37 17

**Review by fqueiruga — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Review by jvz — APPROVED** [review]
Looks nice!

**Comment by daniel-beck** [comment]
Universal approval, so marking `ready-for-merge` so this can go into the upcoming weekly.

**Comment by timja** [comment]
@daniel-beck are you able to resolve the conflicts please?

**Comment by daniel-beck** [comment]
\*cough\* https://github.com/jenkinsci/jenkins/pull/4535#issuecomment-593150643 \*cough\* 😁

**Comment by timja** [comment]
Lol, other one had been updated more recently so it was higher in the list, and your comment got buried above many other comments :p

**Pull request merged**